src/Symfony/Component/HttpFoundation/Session/Storage/Handler/RedisSessionHandler.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\Session\Storage\Handler;
+
+/**
+ * Redis based session storage handler based on the Redis class
+ * provided by the PHP redis extension.
+ *
+ * Direct port of MemcachedSessionHandler to Redis.
+ *
+ * @see http://php.net/redis
+ *
+ * @author Drak <drak@zikula.org>
+ * @author Dalibor Karlović <dkarlovi@gmail.com>
+ */
+class RedisSessionHandler extends AbstractSessionHandler
+{
+    /**
+     * @var \Redis
+     */
+    private $redis;
+
+    /**
+     * @var int Time to live in seconds
+     */
+    private $ttl;
+
+    /**
+     * @var string Key prefix for shared environments
+     */
+    private $prefix;
+
+    /**
+     * List of available options:
+     *  * prefix: The prefix to use for the keys in order to avoid collision
+     *  * expiretime: The time to live in seconds. Defaults to 1 day.
+     *
+     * @param \Redis     $redis   A \Redis instance
+     * @param null|array $options An associative array of options
+     *
+     * @throws \InvalidArgumentException When unsupported options are passed
+     */
+    public function __construct(\Redis $redis, ?array $options = null)
```

Comment:
Why null instead of an empty array?

Reply:
Basically, enables you to omit the default as it's implied, it simplifies the signature and moves the logic away from it.

https://github.com/kalessil/phpinspectionsea/blob/master/docs/code-style.md#non-null-parameters-default-value